internal/release/promote/perform.go

```diff
@@ -116,30 +144,34 @@ func getPromotionMessage(releaseName string, sourceRelease, targetRelease *v1alp
 	return fmt.Sprintf("Promote %s %s -> %s", releaseName, previousVersion, sourceRelease.Spec.Version)
 }
 
-func getBranchName(sourceEnv, targetEnv string, promotedReleaseNames []string) string {
+func getBranchName(info *PromotionInfo) string {
 	var releases string
-	if len(promotedReleaseNames) == 1 {
-		releases = promotedReleaseNames[0]
+	if len(info.Releases) == 1 {
+		releases = info.Releases[0].Name
 	} else {
-		releases = fmt.Sprintf("%d-releases", len(promotedReleaseNames))
+		releases = fmt.Sprintf("%d-releases", len(info.Releases))
 	}
 	uniqueID := uuid.New().String()
-	name := fmt.Sprintf("promote-%s-from-%s-to-%s-%s", releases, sourceEnv, targetEnv, uniqueID)
+	name := fmt.Sprintf("promote-%s-from-%s-to-%s-%s", releases, info.SourceEnvironment.Name, info.TargetEnvironment.Name, uniqueID)
 	if len(name) > 255 {
 		name = name[:255]
 	}
 	return name
 }
 
 // getCommitMessage computes the commit message for the whole promotion operation including all releases
-func getCommitMessage(sourceEnv, targetEnv string, promotedReleaseNames []string, messages []string) string {
-	if len(messages) == 1 {
-		// Put details of single promotion on first and only line
-		return fmt.Sprintf("%s (%s -> %s)", messages[0], sourceEnv, targetEnv)
+func getCommitMessage(pullRequestTemplate string, info *PromotionInfo) (string, error) {
+	tmpl, err := template.New("commit").Parse(pullRequestTemplate)
```

Comment:
It might also make sense to parse the template before entering the command, this way we don't do all this work to realize that the template isn't parsable!